### ISSUE TITLE
Control Ultimate Edition (870780)

### ISSUE THREAD

**Issue by olav-valle** [ValveSoftware/Proton#4157]
# Compatibility Report
- Name of the game with compatibility issues: Control Ultimate Edition
- Steam AppID of the game: 870780

## System Information
- GPU: Nvidia GTX 1080ti
- Driver/LLVM: nvidia 440.100
- Kernel version: 5.4.0-7642-generic
- Link to full system information report as [Gist](https://gist.github.com/olav-valle/081aaccc17c7d1348c19bc2ad05f1e29):
- Proton version: 5.0-9

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-870780.log](https://github.com/ValveSoftware/Proton/files/5138887/steam-870780.log)

## Symptoms 
Generally low performance, though seemingly close to ready for whitelisting.
 
Game is stuttering (seems like a frametiming issue) almost constantly, with frequent hitches (though these are probably caused by shaders still being cached). 

On medium settings at 1440p the performance on a GTX1080ti and i9 9900k is lower than seems reasonable (hovering around 50-55FPS), considering the recommended specs are a GTX 1060 and an i5.

Possibly also experiencing some artifacting in regards to reflections and transparency appearing low res/grainy, but this has not been compared visually to native. The severity of this seemingly comes and goes, and I haven't found a reliable way to reproduce it yet. See screenshots. The silver/white particles on jacket, hair, edges or reflective surfaces of various objects, and entire floor in background of second image. 

Before artifacting:
![870780_20200828003156_1](https://user-images.githubusercontent.com/54496313/91501768-a9504480-e8c6-11ea-9bd4-9e049aa872ec.png)
![870780_20200828003222_1](https://user-images.githubusercontent.com/54496313/91501771-aa817180-e8c6-11ea-8137-d79b308c5bd7.png)

After artifacting:
![870780_20200828000811_1](https://user-images.githubusercontent.com/54496313/91501660-7312c500-e8c6-11ea-9fe9-2a65fa234871.png)
![870780_20200828000823_1](https://user-images.githubusercontent.com/54496313/91501754-9f2e4600-e8c6-11ea-8724-470ebbc47e14.png)

Also, the game really doesn't like being alt-tabbed.

## Reproduction
Click play.

**Comment by McMarius11** [comment]
the performance for me is also not great, please rename in the main game folder Control_DX11.exe to Control_DX12.exe. this fixed the stuttering for me, the game will now use the DX11 exe.

Proton Proton-5.9-GE-5-ST
Distro: Manjaro Linux
Kernel: 5.8.1-3-MANJARO
RAM: 24 GB
GPU Driver: NVIDIA 450.56.06
GPU: NVIDIA GeForce GTX 1070
CPU: Intel Core i5-7600K @ 3.80GHz

**Comment by olav-valle** [comment]
> the performance for me is also not great, please rename in the main game folder Control_DX11.exe to Control_DX12.exe. this fixed the stuttering for me, the game will now use the DX11 exe.

If you rename the DX11.exe, the game won't start because of a 'file not found' error for the DX11.exe file. Are you sure you have this right? Do you mean rename DX11 to DX12, and DX12 to DX11?

**Comment by McMarius11** [comment]
> > the performance for me is also not great, please rename in the main game folder Control_DX11.exe to Control_DX12.exe. this fixed the stuttering for me, the game will now use the DX11 exe.
> 
> If you rename the DX11.exe, the game won't start because of a 'file not found' error for the DX11.exe file. Are you sure you have this right? Do you mean rename DX11 to DX12, and DX12 to DX11?

yes, I renamed the DX11.exe to DX12.exe and DX12.exe to DX11.exe
then the game will use the DX11.exe which is named DX12.exe

it still stutters but its much better then with the DX12.exe

i would recommend installing the Nvidia 450.66 driver
disabling Global Reflection might also help

**Comment by Saancreed** [comment]
I'll never understand why people keep renaming executables (which will probably be undone by every game update), but oh well. In case of Control, you can actually append `-dx11` or `-dx12` to your launch options to bypass the launcher / feature detection logic and force it to immediately start your choice of `Control_DX11.exe` or `Control_DX12.exe` instead.

I believe for this game using d3d11 with DXVK is recommended over d3d12 with vkd3d, but your mileage may obviously vary. Right now DXVK might have some additional stuttering, but should be more stable than vkd3d and over time the stutters will be reduced due to Fossilize kicking in.

**Comment by RiantAnda** [comment]
The issues shown in the screenshots in the first post are also present on Windows with DX12. These are not Proton issues. 
To get ride of those, "Global Reflections" has to be set to "OFF" in the "Display" tab under "Options".

**Comment by sebastian-de** [comment]
The game is playable with `-dx11` and _medium_ graphics settings on a Radeon RX570 here (Arch Linux, linux-zen, mesa-git). Performance could be better, but one of the Valve-hired Mesa developers seems to be on track to improve that:
> Significantly improves performance of a Control compute shader. Also seems
to increase FPS at the very start of the game by ~9% (RX 580, 1080p,
medium settings, no MSAA).

from https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/6558/diffs?commit_id=4baea6ce0921f9060649f98045f2b9db76dc87e2

**Comment by xTeixeira** [comment]
I can't alt tab out of this game. The game window instantly comes back to the top. I'm running GNOME on Xorg, Arch Linux.

Also, the game doesn't launch when using Proton 5.13-1. Proton 5.0-9 works fine except the alt tabbing issue.

**Comment by jrgp** [comment]
I get very good perf for this game using Proton 5.0-9 without any additional changes/command line args. (Except I have mangohud)

FPS: 110-150
Nvidia driver: 455.23.04
GPU: RTX 3080
CPU: AMD 3950x
Ram: 64GB 
Kernel: custom compiled 5.6 
Ubuntu 20.04 

I have not tried alt-tab

**Comment by joshas** [comment]
Having strange issue with this game on AMD RADEON 5600XT, with Proton 5.13-4 and 5.0-10. After varied amount of time (from minutes to hours) game freezes and displays last frame with pixelated streaks of red, green and blue pixels randomly(?) splattered across the screen. Freezes happen not always, got quite a few gaming sessions without any issues. CPU and GPU temperatures are decent, freezes happen on any graphic detail level.
![control-freeze](https://user-images.githubusercontent.com/79141/102383915-c4a9a180-3fd4-11eb-873b-fa2c518c848d.jpg)

Related error messages from system log:

````
kernel: [drm:amdgpu_dm_atomic_commit_tail [amdgpu]] *ERROR* Waiting for fences timed out!
kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx_0.0.0 timeout, signaled seq=7794353, emitted seq=7794355
kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process Control_DX11.ex pid 26902 thread Control_DX:cs0 pid 26941
kernel: amdgpu 0000:0c:00.0: amdgpu: GPU reset begin!
kernel: amdgpu 0000:0c:00.0: [drm:amdgpu_ring_test_helper [amdgpu]] *ERROR* ring kiq_2.1.0 test failed (-110)
kernel: [drm:gfx_v10_0_hw_fini [amdgpu]] *ERROR* KGQ disable failed
kernel: amdgpu 0000:0c:00.0: [drm:amdgpu_ring_test_helper [amdgpu]] *ERROR* ring kiq_2.1.0 test failed (-110)
kernel: [drm:gfx_v10_0_hw_fini [amdgpu]] *ERROR* KCQ disable failed
kernel: [drm:gfx_v10_0_hw_fini [amdgpu]] *ERROR* failed to halt cp gfx
kernel: [drm] free PSP TMR buffer
kernel: amdgpu 0000:0c:00.0: amdgpu: GPU BACO reset
kernel: amdgpu 0000:0c:00.0: amdgpu: GPU reset succeeded, trying to resume
kernel: [drm] PCIE GART of 512M enabled (table at 0x0000008000300000).
kernel: [drm] VRAM is lost due to GPU reset!
kernel: [drm] PSP is resuming...
kernel: [drm] reserve 0x900000 from 0x817e400000 for PSP TMR
````

Has anyone seen anything similar? Any ideas how to figure out what exactly causes this freeze?

**Comment by kisak-valve** [comment]
Hello @joshas, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. Was the game working without issue in the past or have you been seeing this issue as long as you've been playing the game?

**Comment by joshas** [comment]
Here's the [Steam System Info](https://gist.github.com/joshas/904dd8e44e9aefa84f72d9b34ec5f72c) and I got [Proton log](https://gist.github.com/joshas/592aaffc543b8c61a2c7eb02d36dccd2) too, but I cannot find any related error messages in it. 

This is my first play-through of this game, I've purchased it just last week. Looking at reports from other players I expected a crash or two, but now I'm interested, if this issue is specific to my hardware setup only, or not? Only other occasion I've seen similar issue on my system was playing Ori and the Will of the Wisps (1057090), but there was only one or two freezes in total; screen just went full black and similar pixelated pattern appeared. Of course from that time Proton and my OS got few updates.

**Comment by hollisticated-horse** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/4157#issuecomment-746699083

hey there, i've similar problems but in blender for some reason ? same thing, crash, black screen for a few seconds, then a frame with weird artifacts... RX 5700XT

**Comment by hollisticated-horse** [comment]
heya, been having problems with the game... i have a freeze and then a weird window opens and tells me the following :


`File IO failure: readFileFailed, error code 21, filename Z:\mnt\sdb1\steamapps\common\Control\data_packfiles\ep100-000-pc.rmdp `

`File IO failure: openFileFailed, error code 998, filename renderer.ini `

i can cycle between the two with a `retry` and `exit` button 

(i have a few screenshots)

Has happened every time i've played. Other then that, i confirm i also have the particule artifact same as original issue, though maybe less. 

I have changed  from 5.0 to 5.13, same issue.

**Comment by joshas** [comment]
@hollisticated-horse I've been getting same issue (freeze with pixelated frame) in native Linux games too, but got that partially fixed by switching from Wayland to Xorg. Although I still managed to get same freeze, but only once in about 30 hours of total gameplay. Probably video driver issue, as there are issues with same bug descriptions on mesa and amdgpu repositories, so Proton is not at fault here.

**Comment by hollisticated-horse** [comment]
@joshas  yeah probably true, been talking with the guy how maintains the amd linux-firmware ppa, couldn't find an explanation either...

That said, is it better ?

(for my own issue, i'm looking into a faulty ssd :P// could it be that ? )

**Comment by hollisticated-horse** [comment]
okay, so i've installed the game on my hdd, to see if the ssd was at fault for the sudden crash, and i'm getting a :

Wine C++ Runtime Library :
Runtime error!

Program: Z:/home/'user'/.local/share/steam/steamapps/common/Control/Control_DX11.exe abnormal program termination
Press OK to exit the program, or Cancel to start the Wine de bugger.

I'm on Ubuntu 20.10, using proton 5.13. 

help ?

Edit: i'm hearing sound from the game ... the game window is stuck on loading the game. I can access the game menu but not the game itself ...

Edit 2: @kisak-valve  should i start a new issue ?  i'm not using the same GPU and system as the initial post.

**Comment by kisak-valve** [comment]
Hello @hollisticated-horse, in general, we're using one issue report per unofficially supported game title. This is the right place to discuss everything related to Control Ultimate Edition when run with Proton.

Please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

**Comment by hollisticated-horse** [comment]
[steam-870780.log](https://github.com/ValveSoftware/Proton/files/5743970/steam-870780.log)
@kisak-valve  here's the log file. Same error as mentioned above. Anything else that might help ?

**Comment by hollisticated-horse** [comment]
aha, progress !! I think my ssd was dead or dying or just not mounted properly mounted ... So now my game just plain old crashes on an HDD... and still got the weird artifact from before, as in same as the original issue.

i'll try to replicate the crash with `PROTON_LOG=1 %command%`.

any suggestion to help me debug ? i have no clue what i should be working on...

**Comment by hollisticated-horse** [comment]
And got a good old green screen and crash. Still those AMD gpus not functionning correctly

[steam-870780.log](https://github.com/ValveSoftware/Proton/files/5788770/steam-870780.log)

**Comment by axelgenus** [comment]
I can't run this game with Proton 5.13-5 but it work flawlessly with Proton 5.0-10. Performances are not great but it's mostly playable/enjoyable ;)

The regression seems to be related to pressure-vessel because it fails with the following error:
`pressure-vessel-wrap[44654]: openat(/proc/self/fd/21): Too many levels of symbolic links`

Anyone has the same problem? I remember having a similar issue with the Flatpak version and I had to use the Proton community package, built w/o pressure-vessel.

**Comment by hollisticated-horse** [comment]
hey @axelgenus, i don't have that. I'll give `5.0-10` a try. 

I've had a couple of errors with sometimes DirectX11, sometime for no reason crash... Seems to fuck up my drive though, since i need to fsck at reboot because of inode issues...

**Comment by ChemiKyle** [comment]
Game launches on my system, menu works fine. Upon starting a new game, screen remains black, I sometimes get dialogue, and computer freezes, requiring me to power cycle. I have tried a handful of different proton versions (4.11, 5.0, 5.13, experimental, WineHQ staging 6.3 via lutris) and flags, if the game launches at all it results in a total system freeze (though I still get sound if there was sound).

* OS: Debian bullseye/testing (not sid/unstable, as system info reports)
* Kernel: x86_64 Linux 5.10.0-4-amd64
* GPU: AMD R9 Fury
* CPU: AMD Ryzen 2600

Attached is a log using the `-dx11` flag and the output of "Help > System Information" as a `.txt` file.

[steam_system_info.txt](https://github.com/ValveSoftware/Proton/files/6105708/steam_system_info.txt)
[steam-870780.log](https://github.com/ValveSoftware/Proton/files/6105686/steam-870780.log)

___
There is a lot of noise from `388:fixme:d3d12_swapchain_GetFrameStatistics: iface 00000000073bbd40, stats 000000003583fcd8 stub!`

___
My issue likely stems from the Humble Bundle "standard" release not separating DX12 and DX11, there is only DX12 at time of posting: https://steamcommunity.com/app/870780/discussions/0/3103515084616505152/

**Comment by aqxa1** [comment]
DX11 support is back [today](https://store.steampowered.com/news/app/870780/view/3026954526597074908).

**Comment by gadanidis** [comment]
~~Out of the box, the game fails to launch on my machine, apparently due to a failure to load `mscoree.dll`. The attached log is with 5.0-10 but I've tried several different Proton versions (6.3-1, 5.13-6, 5.9-GE-8-ST, and others) and get the same error. Relevant lines from the log:~~

```
511.103:00d4:00d8:err:module:fixup_imports_ilonly mscoree.dll not found, IL-only binary L"Control.exe" cannot be loaded
511.103:00d4:00d8:err:module:LdrInitializeThunk Importing dlls for L"Z:\\home\\tim\\games\\steam\\steamapps\\common\\Control\\Control.exe" failed, status c0000135
```

~~`mscoree.dll` does exist in the prefix so I'm not sure why I'm getting this error. The game itself seems to work fine if I rename `Control_DX11.exe` to `Control.exe`, replacing the original `Control.exe`, so this seems to just be a problem with the launcher or something. (The suggestion above to use `-dx11` or `-dx12` did not work for me, only renaming the files worked.)~~

~~[steam-870780.log](https://github.com/ValveSoftware/Proton/files/6260633/steam-870780.log)~~

Please disregard this, it was just a configuration problem on my end. For some reason I had `mscoree` in `WINEDLLOVERRIDES` in my `.profile`. Not sure when I did that or why but the launcher works fine now that I've removed it.

**Comment by ghost** [comment]
The game  is working fine in DX11 for me. My problem is when i activate DXR function in VKD3D, it was working fine 2 days ago. Now the game doesn't launch with RT.

    OS: Manjaro Gnome 21.0.4
    Kernel: 5.12.2-153-tkg-bmq
    GPU: NVIDIA RTX 3080 (465.27)
    CPU: I7 9770K

[steam-870780.log](https://github.com/ValveSoftware/Proton/files/6488476/steam-870780.log)

New log with WINEDEBUG=+vulkan VKD3D_DEBUG=trace %command%
[steam-870780_debug.log](https://github.com/ValveSoftware/Proton/files/6489223/steam-870780_debug.log)

**Comment by D33M0N** [comment]
Crashes at launch. With error popup that reads:
Error (title)
Error launching the game:
ApplicationName='Control_DX11.exe'
CommandLine=", CurrentDirectory='Z:\home\deemon\.local\share\Steam\steamapps\common\Control', Native error= Internal error.

Manjaro.
Kernel: 5.11.19-1-MANJARO
GPU: Vega64
CPU: R9 5950X

**Comment by D33M0N** [comment]
game started fine, but crashed at some point with ...
```
mai   21 13:03:54 Zen kernel: [drm:gfx_v9_0_priv_reg_irq [amdgpu]] *ERROR* Illegal register access in command stream
mai   21 13:03:54 Zen kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* ring gfx timeout, signaled seq=7025582, emitted seq=7025584
mai   21 13:03:54 Zen kernel: [drm:amdgpu_job_timedout [amdgpu]] *ERROR* Process information: process Control_DX11.ex pid 13387 thread Control_DX:cs0 pid 13487
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x63, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x26, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x61, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed message: 0x37, input parameter: 0x0, error code: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x63, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x26, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed to send message: 0x61, ret value: 0xffffffff
mai   21 13:03:55 Zen kernel: amdgpu: [powerplay] Failed message: 0x37, input parameter: 0x0, error code: 0xffffffff
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
mai   21 13:03:56 Zen kernel: [drm:amdgpu_cs_ioctl [amdgpu]] *ERROR* Failed to initialize parser -125!
... (a lot more of those lines)
```
could be mesa bug though. or kernel or somthing else...
```
mesa-git 21.2.0_devel.139848.307139c7f94-1
kernel: 5.12.2-1-MANJARO
GPU: Vega 64
```

**Comment by hakzsam** [comment]
@D33M0N Is this a recent regression? The "Illegal register access" error is pretty bad.

**Comment by D33M0N** [comment]
> @D33M0N Is this a recent regression? The "Illegal register access" error is pretty bad.

Dunno, Installed Control first time yesterday. Was able to play like half an hour or so fine, until this.

**Comment by hakzsam** [comment]
@D33M0N Is this always reproducible? that error means the driver emits an invalid register, should be easy to fix if I can reproduce myself.

**Comment by D33M0N** [comment]
@hakzsam dunno again, managed to do it second time also. Just when you come out of the first "mind game" floating boxes place with your brand new gun, and start exploring the floor and shooting random things and walls and stuff, then it just happens sooner or later. Previous 2 crashes were with the default proton for Control ... 5.13 I assume. Will try different proton version now.

**Comment by silversquirl** [comment]
The game runs great for me, but strangely seems to steal focus back whenever I try to defocus it. This happens in both full screen and borderless modes, but not windowed mode.
Opening the steam overlay lets me defocus it, but focus will still occasionally flicker to Control and back. The game (and overlay) also seems to freeze sometimes due to being defocused, requiring `pkill -9 Control` - not sure what specifically causes it though.

**Comment by is-gray-wolf** [comment]
I have new NVidia 470.57.02 drivers and I use the latest release and experimental versions of proton. However, the game still does not recognize my 2080Ti as DLSS and RTX enabled. Is there some switch to enable the features now that the driver has them?

**Comment by RiantAnda** [comment]
@the-black-wolf AFAIK you need to enable DLSS and DXR(raytracing) in the "LAUNCH OPTIONS" of the game at the moment:
`VKD3D_CONFIG=dxr %command% -dx12` -> Enables DXR and forces the DX12 version of the game, works for me.
`PROTON_ENABLE_NVAPI=1 %command% -dx12` -> Should enable DLSS, but the game doesn't launch with this option.

**Comment by misyltoad** [comment]
In Proton Experimental I get weird view snapping in Control which doesnt happen in Proton 6.3

I also get weird flickering when playing and the game covers the full screen in fullscreen or borderless (fshack bug perhaps? or compositor?? need to check...)

Does the view snapping happen for anyone else on experimental?

**Comment by misyltoad** [comment]
Flickering appears to be my compositor (KWin) somehow, having a window over the game fixes that.

**Comment by alasky17** [comment]
@Joshua-Ashton I'm not sure exactly what you mean by view snapping?  Could you get a video or describe what you are doing to trigger the behavior?

**Comment by misyltoad** [comment]
Seems to only happen under KDE Wayland actually... Wasn't able to get a video because no screen capturing works there. Hm...

**Comment by kisak-valve** [comment]
Hello @joshas, are you still seeing the issue you reported at https://github.com/ValveSoftware/Proton/issues/4157#issuecomment-746699083? Mesa 20.2.3 is fairly outdated at this point, so it'd be nice if you could retest with a current point release or git main.

**Comment by joshas** [comment]
Kernel Version:  5.13.13-200.fc34.x86_64
Driver:  AMD AMD Radeon RX 5600 XT (NAVI10, DRM 3.41.0, 5.13.13-200.fc34.x86_64, LLVM 12.0.1)
Driver Version:  4.6 (Compatibility Profile) Mesa 21.1.7
Proton 6.3-6
GNOME + Wayland

Played for about an hour, first on medium graphics setting, then tried on maximum. Ran around various locations using end game save, got into battles - could not reproduce the issue. I can only hope, that issue was actually fixed.

**Comment by kisak-valve** [comment]
Thanks for retesting. Between your feedback and https://gitlab.freedesktop.org/mesa/mesa/-/issues/4820, it looks like there's no currently known issues between this game and mesa/RADV, so I'm dropping the corresponding labels.

**Comment by davegorst** [comment]
This previously worked, but now gives me : 

System.ArgumentException: XOR data length expected 2046820352, read 3830

While trying to start up. Tried on 2 different machines - one nVidia/Intel and my main AMD machine. Forcing the DX12 version with -dx12 results in it starting correctly.

**Comment by davegorst** [comment]
Also setting VKD3D_CONFIG=dxr11 is needed to enable raytracing on AMD.

**Comment by gkeep** [comment]
Crashes while launching with DX12 on Fedora 35. I can see the game opening to fullscreen, but it crashes immediately. Tried with Proton Experimental and Proton-6.20-GE-1. Possibly Vulkan related as I have had DOOM Eternal, F1 2020 (DX12) and Cyberpunk 2077 crash on me similarly.

[steam-870780.log (Proton Experimental)](https://github.com/ValveSoftware/Proton/files/7451823/steam-870780.log)
[steam-870780.log (Proton-6.20-GE-1)](https://github.com/ValveSoftware/Proton/files/7451839/steam-870780.log)

Game runs with DX11 with input lag.

Kernel: 5.14.14-300.fc35.x86_64
GPU: NVIDIA GeForce GTX 1650 SUPER
Driver: 470.74.0
Mesa: 21.2.5-1.fc35
DE: KDE Plasma 5.23.2 X11

**Comment by kisak-valve** [comment]
Hello @gkeep, these look like the lines of interest from your log:
```
01a4:info:vkd3d_config_flags_init_once: VKD3D_CONFIG='dxr'.
01a4:info:vkd3d_get_vk_version: vkd3d-proton - applicationVersion: 2.5.0.
01a4:info:vkd3d_instance_init: vkd3d-proton - build: 8ff91b23d61a10c.
01a4:err:vkd3d_create_vk_device: Failed to create Vulkan device, vr -3.
01a4:fixme:hresult_from_vk_result: Unhandled VkResult -3.
```
It might be interesting to test how the game behaves without force enabling dxr with vkd3d-proton. If my memory is correct, nVidia 16xx cards do not have discrete hardware for ray tracing.

**Comment by gkeep** [comment]
> If my memory is correct, nVidia 16xx cards do not have discrete hardware for ray tracing.

You're correct. They are not RTX-capable.

Here are logs without force enabling dxr (launch options are `PROTON_LOG=1 %command% -dx12`):
[steam-870780.log (Proton Experimental)](https://github.com/ValveSoftware/Proton/files/7453801/steam-870780.log)
[steam-870780.log (Proton GE)](https://github.com/ValveSoftware/Proton/files/7453796/steam-870780.log)

EDIT:
Looks like the same vkd3d error output as in previous logs.

Proton Experimental, line 1284
```
01ac:info:vkd3d_get_vk_version: vkd3d-proton - applicationVersion: 2.5.0.
01ac:info:vkd3d_instance_init: vkd3d-proton - build: 8ff91b23d61a10c.
01ac:err:vkd3d_create_vk_device: Failed to create Vulkan device, vr -3.
01ac:fixme:hresult_from_vk_result: Unhandled VkResult -3.
```

**Comment by AlexP11223** [comment]
It seems like the game uses more VRAM than on Windows, which causes issues with the mod fixing blurry textures/texture pop-in on DX12. https://www.reddit.com/r/linux_gaming/comments/qmtlng/blurry_textures_in_dx12_control_fixable_by_mod_on/

Is it expected for VKD3D? Is there any way to optimize VRAM usage?

**Comment by gkeep** [comment]
> Here are logs without force enabling dxr (launch options are `PROTON_LOG=1 %command% -dx12`): [steam-870780.log (Proton Experimental)](https://github.com/ValveSoftware/Proton/files/7453801/steam-870780.log) [steam-870780.log (Proton GE)](https://github.com/ValveSoftware/Proton/files/7453796/steam-870780.log)
> 
> EDIT: Looks like the same vkd3d error output as in previous logs.
> 
> ```
> 01ac:info:vkd3d_get_vk_version: vkd3d-proton - applicationVersion: 2.5.0.
> 01ac:info:vkd3d_instance_init: vkd3d-proton - build: 8ff91b23d61a10c.
> 01ac:err:vkd3d_create_vk_device: Failed to create Vulkan device, vr -3.
> 01ac:fixme:hresult_from_vk_result: Unhandled VkResult -3.
> ```

Looks like I was missing some Vulkan dependent packages. Most notably `xorg-x11-drv-nvidia-cuda` which fixed my issue with Vulkan and DX12 in Control, Doom Eternal and F1 2020 (DX12).

**Comment by kisak-valve** [comment]
Control Ultimate edition Stuck in mission = cant finish game

Issue transferred from https://github.com/ValveSoftware/Proton/issues/5994.
@mushroomeo posted on 2022-07-14T13:33:12:

# Compatibility Report
- Name of the game with compatibility issues: Control Ultimate edition
- Steam AppID of the game: 870780

## System Information
- GPU: RTX 3080 Ti
- Driver/LLVM version: NVIDIA Driver Version: 515.57
- Kernel version: 5.18.11-xanmod-1-1, but tryed normal kernal too and didnt work
- Link to full system information report: https://gist.github.com/mushroomeo/dff5944a3a2c1b8f0d3da1a30dbaaaa6
- Proton version: All and additionally some glorious eggroll

## I confirm:
- [ ] that I haven't found an existing compatibility report for this game.
- [ ] that I have checked whether there are updates for my system available.

PROTON LOG:
[steam-870780.log](https://github.com/ValveSoftware/Proton/files/9112177/steam-870780.log)


## Symptoms 
getting stuck by talking with npc. so cant progress/finish game
i can rorate camera and and go in menus but nothing happening else. the npcs just standing and no doalogue. = stuck

## Reproduction
just do the mission: my borthers keeper, find dylan. and talk to the npc after running through white bridge...

https://user-images.githubusercontent.com/9019563/178994150-e9350a8e-2659-4220-b0a7-a1fd0570db73.mp4

![b7useyl](https://user-images.githubusercontent.com/9019563/178994493-4d8a6975-5c08-44f5-9073-9a65b56f4af0.jpg)


trying verify integrity of game , tryed different protons, tryed find in internet about if someone got same bug, nothing seems to work , its proton or game bug i cant test since i dont have windows OS anymore.

**Comment by Calinou** [comment]
On i9-13900K + RTX 4090 (NVIDIA 530.41.03), stuttering still seems very present on the latest Proton Experimental, both in DX11 and DX12 mode (with or without raytracing or DLSS). This is in DX11 mode for instance, trying to reach 120 FPS V-Sync while looking at a wall:

![image](https://user-images.githubusercontent.com/180032/235269655-21fceaa5-4f8d-4dc0-97c1-82f659517138.png)

Disabling V-Sync doesn't improve the situation. This doesn't appear to be shader compilation stutter, as it remains persistent for several minutes in the intro sequence (even if I stand still and do nothing). Even setting the game to its absolute lowest settings and 960×540 render resolution still doesn't yield a stable 120 FPS in DX11 mode (it's as unstable as in the graph shown above).

Note that I'm also using [this patch](https://community.pcgamingwiki.com/files/file/2581-control-hdrultrawidedlssrt-patch/).

**Edit:** On the same PC, the game runs very smoothly on Windows 11, even with the patch and raytracing enabled.

**Comment by mrmatteastwood** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/4157#issuecomment-1528202729

Experiencing constant stuttering as well on i9-12950HX and RTX 3080 Ti laptop, playing in 1080p with all settings maxed out including RTX. Latest Proton Experimental or ProtonGE 7.55. Nvidia driver 525.105.17.

The game seems to always hover at just under 60 FPS, somewhere between 55 and 59. When I enable DLSS and pick any rendering resolution, even the lowest, the FPS hover at around 50-55.

**Comment by mrmatteastwood** [comment]
Quick follow-up: also happens with Nvidia driver 530.41.03. The most severe stuttering I encounter feels like the kind of stuttering when the game loads something in the background. It even freezes sometimes for a second.

**Comment by kisak-valve** [comment]
Control: Ultimate Edition [870780]

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7201.
@Saijin-Naib posted on 2023-10-26T02:43:26:

# Compatibility Report
- Name of the game with compatibility issues: Control: Ultimate Edition
- Steam AppID of the game: 870780

## System Information
- GPU: ASRock Intel Arc A380 6GB
- Video driver version: Mesa 23.1.9
- Kernel version: 6.5.9
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: Proton Experimental, Build 12481161

## I confirm:
- [X ] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-870780.log](https://github.com/ValveSoftware/Proton/files/13172422/steam-870780.log)

## Symptoms 
None of the DLC/Content associated with Control: Ultimate Edition is available within Steam for Linux, so I am missing the AWE DLC and the Foundation DLC.

## Reproduction
1) Run game in Steam for Linux
2) Observe that the Ultimate Edition Content is not present

## GISTs for System Information & Steam Runtime Information
https://gist.github.com/Saijin-Naib/47cc8edc0bc31a5110c3a0bb7bb0f144

**Comment by lelgenio** [comment]
Replying to https://github.com/ValveSoftware/Proton/issues/4157#issuecomment-1537439286

I'm experiencing this on an all-AMD machine too. It's due to an update on linux/mesa/something else, not proton. For now I rolled back the system and it's fine, when I get more time I'll try to figure out what is causing this. Nixos 23.11 (commit 2d2ea8e) is fine, Nixos unstable has the issue.

@mrmatteastwood, did you experience this issue in any other game? I may or may not have it on DayZ too.

**Comment by lelgenio** [comment]
Update on my previous comment. With the latest version of everything the "loading stutters" are gone! :smile:

**Comment by Rodancoci** [comment]
> It seems like the game uses more VRAM than on Windows, which causes issues with the mod fixing blurry textures/texture pop-in on DX12. https://www.reddit.com/r/linux_gaming/comments/qmtlng/blurry_textures_in_dx12_control_fixable_by_mod_on/
> 
> Is it expected for VKD3D? Is there any way to optimize VRAM usage?

I am experiencing this in 2025. At random points during gameplay textures stop streaming correctly because of this, leading to blurry textures even with 8GB of VRAM.

**Comment by NyaomiDEV** [comment]
The HDR toggle is not available on KDE Plasma 6.3 using an HDR compatible monitor and having set the DE to use HDR. Using Proton 9.0-4 and also Experimental. AMD 6900XT

**Comment by D33M0N** [comment]
> The HDR toggle is not available on KDE Plasma 6.3 using an HDR compatible monitor and having set the DE to use HDR. Using Proton 9.0-4 and also Experimental. AMD 6900XT

This is not, how you can use HDR in linux. yet. You need to jump through extra hoops, like using HDR compatible compositor, like gamescope. https://wiki.archlinux.org/title/Gamescope#HDR_support
For example me running elden ring with HDR is like this with steam command line:
`DXVK_HDR=1 gamescope -e -f -W 3840 -H 2160 --hdr-enabled --adaptive-sync -- %command%`

**Comment by zierf** [comment]
KDE is HDR compatible as long as the monitors support it. I can activate HDR in _Control_.
I just didn't calibrate it perfectly and had to downsize the screenshots for GitHub.

- _Heroic Games Launcher_
- _GE-Proton9-27_
- _DXVK-NVAPI_ enabled
- <ins>disabled</ins> _gamescope_

<details>
<summary>KDE Settings</summary>

![KDE Settings](https://github.com/user-attachments/assets/3a87fba3-c92e-4584-ba11-593759c89f49)
</details>

Try to set `DXVK_HDR=1` as environment variable to make Proton support HDR in games.

![DXVK_HDR](https://github.com/user-attachments/assets/5bc83eb3-2049-4b51-a26b-957cc5a62198)

![Launch Parameters](https://github.com/user-attachments/assets/4870cfcd-5b14-48a9-a6ae-70e08929d635)

---

__HDR:__ _on_
![Control HDR](https://github.com/user-attachments/assets/20ae52d7-c9d6-4317-bcbc-0b9545b7b272)

__HDR:__ _off_
![Control SDR](https://github.com/user-attachments/assets/96e3a86b-31d6-4a09-b988-bfe0be030803)

**Comment by ranplayer** [comment]
I've found a minor issue with Control for any Proton version >= 8.0-5: if you restart the game after enabling vsync and fullscreen, it will crash on the main menu (after pressing any action button on the controller). DX11 mode (`-dx11`) has the same issue.

Logs
- [steam-870780-proton10.tar.gz](https://github.com/user-attachments/files/21582310/steam-870780-proton10.tar.gz)
- [steam-870780-proton9.tar.gz](https://github.com/user-attachments/files/21582316/steam-870780-proton9.tar.gz)

Specs:
- OS: Arch
- Kernel: 6.12.41
- GPU: RX 6700 XT (Mesa 25.1.7)
- CPU: Ryzen 5 5700G (9c/16t)
- DE: Plasma 6.4.3 (X11 and Wayland)

**Comment by alasky17** [comment]
@ranplayer We've been unable to reproduce this issue.  I wonder if this is also caused by some unusual resolution similar to the theory for https://github.com/ValveSoftware/Proton/issues/4896#issuecomment-3127142864.  Uploading the output of xrandr could be helpful here.  

If you are able to also copy your system information from Steam (Steam -> Help -> System Information and Steam -> Help -> Steam Runtime Diagnostics) and put each in a gist, then include a link to the gists in this issue report, that might also help shed light onto why you are hitting this regression with Proton versions newer than 8.0 on your system.

**Comment by ranplayer** [comment]
@alasky17 I use the default/max device resolution of 1080p with the refresh rate set to 75hz (max is 144hz). I'm attaching the requested logs:
- [xrandr.tar.gz](https://github.com/user-attachments/files/22215153/xrandr.tar.gz)
- [steam_report-2025_09_08.tar.gz](https://github.com/user-attachments/files/22215152/steam_report-2025_09_08.tar.gz)

<img width="437" height="206" alt="Image" src="https://github.com/user-attachments/assets/2f7539f7-652a-4bc7-93a6-243eb8e21ba6" />

**Comment by ranplayer** [comment]
@alasky17 I've just retested with Proton 10-3 (RC) and updated kernel (6.12.53) and graphics (Mesa 25.2.5) and got the same issue.

Logs:[steam-870780.tar.gz](https://github.com/user-attachments/files/23026082/steam-870780.tar.gz)

~~Limiting the resolutions to 32, like the fix for [Ultimate Marvel vs Capcom 3](https://github.com/ValveSoftware/Proton/issues/1946#issuecomment-3403052119) , worked!~~
```
elif os.environ.get("SteamGameId", 0) in [
                    "524220", #NieR: Automata
                    "814380", #Sekiro: Shadows Die Twice
                    "374320", #DARK SOULS III
                    "870780", # Control: Ultimate Edition
                    ]:
self.env["PROTON_LIMIT_RESOLUTIONS"] = "32"
```
Edit: just retested now and got the crash again. After trying some things I believe it might be related to SteamInput (I'm using a DS4): if instead of pressing X (DS4) on the menu, I press "Return" (keyboard), I don't get the crash (also if I disable SteamInput... although I lose controller support)

**Comment by Calinou** [comment]
I've tried to enable HDR on Control, but it looks blown out (until I pause the game, in which case it applies a dark overlay to the whole scene). Any ideas?

I'm using https://github.com/Zamundaaa/VK_hdr_layer from the Fedora repositories since it appears to be required on NVIDIA. It seems to work well in Quake 2 RTX.

Launch options:

```
PROTON_ENABLE_WAYLAND=1 PROTON_ENABLE_HDR=1 ENABLE_HDR_WSI=1 DXVK_HDR=1 %command%
```

System information:

```
Operating System: Fedora Linux 42
KDE Plasma Version: 6.4.5
KDE Frameworks Version: 6.19.0
Qt Version: 6.9.2
Kernel Version: 6.17.4-200.fc42.x86_64 (64-bit)
Graphics Platform: Wayland

Processors: 32 × AMD Ryzen 9 9950X3D 16-Core Processor
Memory: 64 GiB of RAM (62.3 GiB usable)
Graphics Processor: NVIDIA GeForce RTX 5090 (driver 580.95.05)
Manufacturer: Micro-Star International Co., Ltd.
Product Name: MS-7E49
System Version: 1.0
```

Using Proton Experimental. HDR is enabled in the system settings (I'm not using Gamescope). I did try Gamescope from the Fedora repositories, but it segfaults on startup on Control (it works on vkcube with `--hdr-enabled`).

PS: On the bright side, https://github.com/ValveSoftware/Proton/issues/4157#issuecomment-1528202729 is completely resolved now 🙂

**Comment by zierf** [comment]
@Calinou There is no _Wine-Wayland_ support in any official Proton version and _Proton Experimental_ is one of them. 
You'll either need to try it with Gamescope or use a Proton version with Wine-Wayland support.
Your Launch Options should work for the latter.

---

HDR with Nvidia GPUs doesn't seem to be fully supported by the driver.
Dark areas are far too dark and light areas are far too bright.

<details>
<summary>HDR enabled (broken)</summary>

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/d19bf869-1511-47e2-8e93-2b7b75414125" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/c1b0fe6c-42bf-490c-ac7f-04c870bc5a77" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b51e2b9b-28a3-48e9-bf74-70b0feee3603" />
</details>

I've managed to get it working, currently using `GE-Proton10-25`.

<details>
<summary>HDR enabled (fixed)</summary>

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/bed6e650-dcbd-4dba-87bd-c87e905ba8f7" />

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/7518986d-cfb4-42a1-8308-396382a8d84c" />
</details>

The easiest way is to use HDR together with a Proton version that supports _Wine-Wayland_.
Unfortunately, the Steam overlay does not yet support _Wayland_ and therefore can no longer be opened in game.

```SH
PROTON_ENABLE_WAYLAND=1 PROTON_ENABLE_HDR=1 ENABLE_HDR_WSI=1 MANGOHUD=1 gamemoderun %command%
```

Alternatively, HDR can be used via Gamescope. This still allows the use of the Steam overlay.
I had to install a `gamescope-wsi` package for the Gamescope `WSI` layer, but it may already be included on most systems.

```SH
ENABLE_HDR_WSI=1 gamescope -f -r 144 -o 30 -W 3840 -H 2160 --mangoapp --hdr-enabled --hdr-debug-force-output -- env ENABLE_GAMESCOPE_WSI=1 DISABLE_HDR_WSI=1 PROTON_ENABLE_HDR=1 gamemoderun %command%
```

I currently have the following launch options for _Control_.

```SH
OBS_VKCAPTURE=1 GAMESCOPE_WSI_HIDE_PRESENT_WAIT_EXT=1 ENABLE_HDR_WSI=1 gamescope -f -r 144 -o 30 -W 3840 -H 2160 --mangoapp --adaptive-sync --hdr-enabled --hdr-debug-force-output -- env ENABLE_GAMESCOPE_WSI=1 DISABLE_HDR_WSI=1 PROTON_ENABLE_HDR=1 gamemoderun %command% -dx12
```

__A few Notes:__
- Only Nvidia GPUs should need `ENABLE_HDR_WSI` (because of the incomplete HDR driver)
  - install `vk-hdr-layer-kwin6` / `vulkan-hdr-layer-kwin6`, depending on your system
  - you can find the source at [Zamundaaa/VK_hdr_layer](https://github.com/Zamundaaa/VK_hdr_layer)
- AMD GPUs only need it with [mesa prior to version 25](https://wiki.archlinux.org/title/HDR_monitor_support#Vulkan_HDR_WSI)
-  `--hdr-debug-force-output` should no longer be needed nowadays, but I needed it for a proper image with Nvidia
- replace the values for FPS cap and monitor resolution in Gamescope with proper values for your own system.

See also [this Reddit post](https://www.reddit.com/r/linux_gaming/comments/1k9mqat/comment/mplg3ql/), which pointed me to the only working solution I could find.
It also briefly mentions settings for _Heroic Games Launcher_.

If anyone happens to be using Steam Flatpak and has managed to get it working, some gamers would surely appreciate a short comment in [their Issue #1308](https://github.com/flathub/com.valvesoftware.Steam/issues/1308#issuecomment-3539406620).

**Comment by FasterFall** [comment]
At start game would just give this error but then work fine, but after playing a while and now trying to relaunch the game, the game's blackscreen rapidly flashes on and off, and I have to logout and log back in into the KDE session, as even closing the game doesn't seem to make KDE usable again.

After testing, it appears this happens (the making KDE unsable part) when "fullscreen" instead of "borderless" is selected. This has to be selected however, otherwise the game just doesn't minimize.

Either way, even when that doesn't happen, this error always pops up (but the game launches fine otherwise when this happens):

"We could not restore your progress on locked achievements. All previously unlocked achievements, rewards and saved game progress are unaffected by this. If you proceed now, your progress on locked achievements will be reset."

This is the GOG version. Adding the game on steam as a non-steam game also causes this issue as well as using GE-Proton.

It appears the error is a [bug](https://www.gog.com/forum/control/patch_13_errors_and_advertising) for Control and could happen on Windows as well, however, the other issue mentioned here (inability to alt-tab out when on borderless and has to be switched to fullscreen, but next time it launches with that, the game basically freezes and flickers and makes KDE unusable)

Edit: So the way to run this game (edit 2: without issues) is using wine-wayland, GE-Proton, and on borderless (fullscreen does make the game still hang when launching, but it doesn't flicker and bring the whole DE down with it, also if enabled after launch the KDE taskbar becomes visible, so it has to be borderless).

**Comment by conveyorBeltOverlord** [comment]
@FasterFall did you find any solutions for the achievements error? im having that error on windows and game runs fine but some of the achievements doesnt work (cleaning 5 control point and collecting 100 documents doesnt progress) but some do work (cleaning 10 control point weirdly)

**Comment by FasterFall** [comment]
@conveyorBeltOverlord No. As I said it appears to be a bug for the GOG (maybe Steam too) version of Control. I don't know what exactly causes it, but the Control devs (or GOG) would need to fix it. I don't know if it happens on all setups, there might be some workaround but I haven't looked that far into it.
